wrappers/golang/curves/bls12377/tests/poseidon_test.go

```diff
@@ -0,0 +1,42 @@
+package tests
+
+import (
+	"testing"
+
+	core "github.com/ingonyama-zk/icicle/v2/wrappers/golang/core"
+	cr "github.com/ingonyama-zk/icicle/v2/wrappers/golang/cuda_runtime"
+	bls12_377 "github.com/ingonyama-zk/icicle/v2/wrappers/golang/curves/bls12377"
+	poseidon "github.com/ingonyama-zk/icicle/v2/wrappers/golang/curves/bls12377/poseidon"
+)
+
+func TestPoseidon(t *testing.T) {
+
+	arity := 2
+	numberOfStates := 1
```

Comment:
Is this sufficient to show that the test is testing what we want?

Reply:
well I would have liked to test for different length and arities but I couldn't find an implementation to check against that does everything. Especially for numberOfStates it's hard to find a library where it's not set to 1.
I'll add more tests later on, just need to find a library (in another language most likely) that support several curves and parameters